### PR TITLE
Add a tiny-glibc image

### PR DESCRIPTION
**Pull request by luxas**
Please review and push: @girishkalele 

cc @bprashanth

### CODE REVIEW

**Comment by girishkalele** [comment]
Ah I see - the base image I was using is missing glibc - I thought since I was using the same base image as the pause.c pause image, it would be compatible for dnsmasq C code and contain the needed C libraries.

File: tiny-glibc/Makefile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	docker run -it --rm -v $(TEMP_DIR):/build $(BIND_QEMU_VOLUME_ARG) $(BUILDIMAGE) /bin/bash -c \
+		"apt-get update \
+		&& apt-get download libc6 \
+		&& mkdir libc \
+		&& dpkg-deb -x *.deb libc \
+		&& cd libc \
+		&& tar -czf /build/glibc.tar.gz ."
+
+	docker build -t $(REGISTRY)/$(IMAGE)-$(ARCH):$(TAG) $(TEMP_DIR)
+
```
Comment (girishkalele) [inline tiny-glibc/Makefile]: And wipe out the temp directory here ? And probably after all targets (since the mktemp runs for all invocations).

**Comment by girishkalele** [comment]
@luxas 

Not familiar with this cross-compilation stuff (is it running a qemu emulator inside Docker containers ?) but apart from my comment on the temp directory cleanup, this LGTM.

Just curious, do you have a multi-architecture e2e test framework ?
Perhaps using QEMU emulators to emulate the master and minion nodes ?

**Comment by luxas** [comment]
Updated. Please push.
I've documented things in https://github.com/kubernetes/kubernetes/pull/26863

**Comment by thockin** [comment]
Can you explain WHY in more detail?  This looks pretty brittle, and I *really* don't want to be in the business of maintaining a base-image in light of security updates and so forth.

As an alternative, there is already `busybox:glibc` on docker hub.  Can we start from there?  Can we convince THEM to build this for multi-arch automatically?

**Comment by luxas** [comment]
WTF!!! I hadn't noticed there are `glibc` images already cross-compiled on the Hub!

Why: We must have small cross-compiled images with glibc (for dynamic linking) for dnsmasq, probably minikube/localkube/hyperkube and in some other places as well, but it now seems we've found the base images.

Thanks for questioning, I'll delete the folder directly. I just was thinking about helping #1125 as fast as possible.

**Pull request merged**